src/Core/Domain/Manufacturer/Query/GetManufacturerForViewing.php

```diff
@@ -0,0 +1,72 @@
+<?php
+/**
+ * 2007-2019 PrestaShop and Contributors
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2019 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+namespace PrestaShop\PrestaShop\Core\Domain\Manufacturer\Query;
+
+use PrestaShop\PrestaShop\Core\Domain\Language\ValueObject\LanguageId;
+use PrestaShop\PrestaShop\Core\Domain\Manufacturer\ValueObject\ManufacturerId;
+
+/**
+ * Get manufacturer information for viewing
+ */
+class GetManufacturerForViewing
+{
+    /**
+     * @var ManufacturerId
+     */
+    private $manufacturerId;
+
+    /**
+     * @var LanguageId Language in which manufacturer is returned
+     */
+    private $languageId;
+
+    /**
+     * @param int $manufacturerId
+     * @param int $languageId
+     */
+    public function __construct($manufacturerId, $languageId)
```

Comment:
@matks I've passed `$languageId` into `Command`, this way command handler becomes context independent and in fact, it makes sense to pass language to command, wdyt?

Reply:
Could you discuss it with Pablo? Since he's the most experienced in this stuff.

Reply:
I lack experience on the whole CQRS implementation, but it seems better this way
What is the alternative? to inject the context in the constructor? or in the handler?

Reply:
> What is the alternative? to inject the context in the constructor? or in the handler?

It's not an alternative, but current implementation. :smile:

Actually it's not even related to CQRS. Injecting context data into any service **shouldn't** be done via `__constructor`, but rather via method call.

Reply:
I was thinking, could it be that the `ManufacturerId` contains the languageId as well? it's like a double ID for a resource which makes sens for a multi lang resource

Reply:
No, I don't think so, `LanguageId` is not part of Manufacturer identity. It would be like having your _Personal Code_ related to the languages you can speak, which wouldn't make sense, right?

Reply:
Alright